### ISSUE TITLE
Bug: Found unparsable segment @L003P037: 'group (order by o_or...'

### ISSUE THREAD

**Issue by HassanJaveed84** [alanmcruickshank/sqlfluff#293]
Running `sqlfluff lint`  version 0.3.1 on the following sql:
```sql
SELECT ARRAY_AGG(o_orderkey) WITHIN GROUP (ORDER BY o_orderkey ASC)
  FROM orders
```
```
L:   1 | P:  37 |  PRS | Found unparsable segment @L001P037: 'group (order by o_or...'
```

**Comment by alanmcruickshank** [comment]
What dialect of SQL are you running? That looks a little like Redshift to me, but want to make sure.

**Comment by alanmcruickshank** [comment]
Looks like it's definitely within [redshift](https://docs.aws.amazon.com/redshift/latest/dg/r_LISTAGG.html), but also part of [postgres](https://www.postgresql.org/docs/current/functions-aggregate.html), so I'm happy to add this to the postgres dialect which has been recently added. I'll put a PR together and this should be in the next release this weekend.

Thanks for reporting it!

**Comment by HassanJaveed84** [comment]
@alanmcruickshank we're using snowflake. That example is also from snowflake docs, but you're right it's also part of postgres. However, I have not tested by passing `--dialect snowflake` as it's not in 0.3.1 release. Can we also add it to snowflake dialect ?

**Comment by alanmcruickshank** [comment]
Yep - the snowflake dialect inherits from the postgres one (and the redshift one will inherit from there if we implement that one too), so by implementing this in the postgres one it should be available for snowflake too.

**Comment by HassanJaveed84** [comment]
Great, Thank you!
